Cargo.toml

```diff
@@ -37,4 +38,22 @@ debug-assertions = true
 overflow-checks = true
 
 [workspace.dependencies]
```

Comment:
In `miden-base` and other crates we've started to group dependencies by their categories (e.g., see [here](https://github.com/0xMiden/miden-base/blob/next/Cargo.toml#L40). Maybe makes sense to do this here as well?

Reply:
Sure!